crates/bevy_ecs/src/schedule_v3/schedule.rs

```diff
@@ -0,0 +1,1001 @@
+        let label = label.dyn_clone();
+        match self.inner.get_mut(&label) {
+            Some(container) => match container.take() {
+                Some(schedule) => Ok(schedule),
+                None => Err(ExtractionError::AlreadyExtracted(label)),
+            },
+            None => Err(ExtractionError::Unknown(label)),
+        }
+    }
+
+    /// Re-inserts the schedule corresponding to the label into the map.
+    ///
+    /// The schedule must have been previously extracted using [`check_out`](#method.check_out).
+    ///
+    /// # Errors
+    ///
+    /// TODO
+    pub(crate) fn check_in(
+        &mut self,
+        label: &dyn ScheduleLabel,
+        schedule: Schedule,
+    ) -> Result<(), ExtractionError> {
+        let label = label.dyn_clone();
+        match self.inner.get_mut(&label) {
+            Some(container) => match container.take() {
+                Some(_) => Err(ExtractionError::NotExtracted(label)),
+                None => {
+                    *container = Some(schedule);
+                    Ok(())
+                }
+            },
+            None => Err(ExtractionError::Unknown(label)),
+        }
+    }
+
+    /// Iterates all system change ticks and clamps any older than [`MAX_CHANGE_AGE`](crate::change_detection::MAX_CHANGE_AGE).
+    /// This prevents overflow and thus prevents false positives.
+    ///
+    /// **Note:** Does nothing if the [`World`] counter has not been incremented at least [`CHECK_TICK_THRESHOLD`](crate::change_detection::CHECK_TICK_THRESHOLD)
+    /// times since the previous pass.
+    pub(crate) fn check_change_ticks(&mut self, change_tick: u32, last_check_tick: u32) {
+        #[cfg(feature = "trace")]
+        let _span = bevy_utils::tracing::info_span!("check ticks").entered();
+        for schedule in self.inner.values_mut().flatten() {
+            schedule.check_change_ticks(change_tick, last_check_tick);
+        }
+    }
+}
+
+/// TBD
+pub struct Schedule {
+    graph: ScheduleMeta,
+    executable: SystemSchedule,
+    executor: Box<dyn SystemExecutor>,
+}
+
+impl Schedule {
+    pub fn new() -> Self {
+        Self {
+            graph: ScheduleMeta::new(),
+            executable: SystemSchedule::new(),
+            executor: Box::new(MultiThreadedExecutor::new()),
+        }
+    }
+
+    pub fn add_system<P>(&mut self, system: impl IntoSystemConfig<P>) {
+        self.graph.add_system(system);
+    }
+
+    pub fn add_systems<P>(&mut self, systems: impl IntoSystemConfigs<P>) {
+        self.graph.add_systems(systems);
+    }
+
+    pub fn configure_set(&mut self, set: impl IntoSystemSetConfig) {
+        self.graph.configure_set(set);
+    }
+
+    pub fn configure_sets(&mut self, sets: impl IntoSystemSetConfigs) {
+        self.graph.configure_sets(sets);
+    }
+
+    pub fn set_default_set(&mut self, set: impl SystemSet) {
+        self.graph.set_default_set(set);
+    }
+
+    pub fn set_executor(&mut self, executor: ExecutorKind) {
```

Comment:
I think this API is too constrained: advanced users should be able to pass in custom executors.

Reply:
The three executor types are tied to the specific implementation of `SystemSchedule`, which is not `pub`. We're not just handing a slice of systems anymore. What should a generic API look like and where does it fit in?

Reply:
Ah, right. Nope, I see your point for now: this isn't worth biting off right now.